### PR TITLE
Fix default sampling option - should be `AllDataAndRecorded`

### LINKED ISSUE #67: ASP.NET Core instrumentation isn't producing traces by default
Hard to come up with a fitting title, i've been experimenting with this stuff because all of the blog posts made me excited but i'm not sure if i'm holding it right. After upgrading seq, upgrading nugets and installing the SerilogTracing libs + adding
```csharp
            using var _ = new ActivityListenerConfiguration()
                .Instrument.AspNetCoreRequests()
                .Instrument.SqlClientCommands()
                .TraceToSharedLogger();
```
as well as changing `WriteTo.Seq(...)` to `WriteTo.SeqTracing` i've noticed no change, other than the log events now have a `Trace (...)` option in seq. Clicking on that and using `Show` brings up nothing without any explanation as to why.
image
digging around in the API requests i found this query text being sent that comes back empty handed
```sql
select coalesce(@Resource.service.name, Application, 'unknown_service') as serviceName, 
        @Message as message, 
        @Level as level, 
        @TraceId as traceId, 
        @SpanId as spanId, 
        @ParentId as parentId, 
        TotalMilliseconds(@Elapsed) as elapsedMs,
        ToIsoString(@Start) as start, 
        ToIsoString(@Timestamp) as timestamp,
        @Start as startTick,
        @Timestamp as timestampTick
        
        from stream 
        where Has(@Start) and Has(@SpanId) and @TraceId = '8bcad354512a096b58a7adb8cc051597'
```
playing around with that i've figured that the reason for this is that none of the events have the magic `@Start` property, whose presence or absence you can see nowhere in seq as far as i could tell.
I did expect to get some builtin tracing, especially after enabling Instrumentation for SqlClient and ASP.NET Core but in lack thereof, i went ahead and added `using var activity = Log.Logger.StartActivity("Selecting Verträge");`  to a random API.
Now i do get something in the trace view, even builtin ones
image
not sure why `SQL SELECT` is not being printed without me having to add `StartActivity`.
Also not sure why this says missing 
image
I went looking in the example apps to check if i did miss something obvious but didn't really find anything. I was confused that the example app configures both SeqTracing and OpenTelemetry, this is not necessary, is it?
https://github.com/serilog-tracing/serilog-tracing/blob/b5f5a490758b3b3f4c3d6786e450a810b0ffca87/example/Example.WeatherService/Program.cs#L12-L14
I've also been using `App` consistently across all services for the application name, saving on those 8 bytes ('App' vs. 'Application') for every log, i guess i'll have to rename that now?

### PR DESCRIPTION
**Pull request by nblumhardt**
An early mistake here led to some brittle, defensive code, which should now be gone :-)

**TL;DR** is that when configuring ASP.NET Core instrumentation, if you're building an internal service, choose `opts.IncomingTraceParent = IncomingTraceParent.Trust`, if it's external use `IncomingTraceParent.Ignore`, and if you are unsure, leave it as the default `IncomingTraceParent.Accept`, with some caveats when sampling is not configured deterministically.

Fixes #67

### CODE REVIEW

**Review by KodrAus — APPROVED** [review]
Looks good to me! This cleans things up nicely :+1:
